allennlp/data/iterators/basic_iterator.py

```diff
@@ -17,21 +20,32 @@ class BasicIterator:
         The size of each batch of instances yielded when calling the iterator.
     """
     def __init__(self, batch_size: int = 32):
-        self.batch_size = batch_size
+        self._batch_size = batch_size
 
+    @overrides
     def __call__(self, dataset: Dataset):
-        grouped_instances = self._create_batches(dataset)
-        self.last_num_batches = len(grouped_instances)  # pylint: disable=attribute-defined-outside-init
-
-        padding_lengths = dataset.get_padding_lengths()
         while True:
-            for group in grouped_instances:
-                batch = Dataset(group)
-                yield batch.as_arrays(padding_lengths, verbose=False)
+            for batch in self.yield_one_pass(dataset):
```

Comment:
Huh, I wasn't aware of the `yield from` syntax, and got sniped looking up coroutines...